schemas/json/layout/layout.schema.v1.json

```diff
@@ -0,0 +1,213 @@
+            "description": "Boolean value indicating if the component should be presented as read only. Defaults to false."
+          },
+          "textResourceBindings": {
+            "type": "object",
+            "title": "textResourceBindings",
+            "description": "Text resource bindings for a component.",
+            "additionalProperties": {
+              "type": "string"
+            }
+          },
+          "dataModelBindings": {
+            "type": "object",
+            "title": "dataModelBindings",
+            "description": "Data model bindings for a component.",
+            "additionalProperties": {
+              "type": "string"
+            }
+          }
+        },
+        "required": ["id", "type"],
+        "allOf": [
+          { "if": {"properties": {"type": { "const": "Datepicker"}}}, "then": { "$ref": "#/definitions/datepickerComponent"}},
+          { "if": {"properties": {"type": { "const": "FileUpload"}}}, "then": { "$ref": "#/definitions/fileUploadComponent"}},
+          { "if": {"properties": {"type": { "const": "NavigationButtons"}}}, "then": { "$ref": "#/definitions/navigationButtonsComponent"}},
+          { "if": {"properties": {"type": { "const": "Group"}}}, "then": { "$ref": "#/definitions/groupComponent"}},
+          { "if": {"properties": {"type": { "const": "Dropdown"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "RadioButtons"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "Checkboxes"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "AddressComponent"}}}, "then": { "$ref": "#/definitions/addressComponent"}}
+        ]
+      },
+      "fileUploadComponent": {
+        "properties": {
+          "maxFileSizeInMB": {
+            "title": "maxFileSizeInMB",
+            "description": "Sets the maximum file size allowed in megabytes.",
+            "type": "integer"
+          },
+          "maxNumberOfAttachments": {
+            "title": "maxNumberOfAttachments",
+            "description": "Sets the maximum number of attachments allowed to upload.",
+            "type": "integer"
+          },
+          "minNumberOfAttachments": {
+            "title": "minNumberOfAttachments",
+            "description": "Sets the minimum number of attachments to upload",
+            "type": "integer"
+          },
+          "displayMode": {
+            "title": "displayMode",
+            "description": "Sets the display mode for the file upload component.",
+            "type":"string",
+            "enum": ["simple", "list"]
+          },
+          "hasCustomFileEndings": {
+            "title": "hasCustomFileEndings",
+            "description": "Boolean value indicating if the component has valid file endings",
+            "type": "boolean"
+          },
+          "validFileEndings": {
```

Comment:
Add the most common file types as `examples`. Like pdf, cvs, etc.
https://json-schema.org/understanding-json-schema/reference/generic.html#annotations